micromagnetictests/calculatortests/zhangli.py

```diff
@@ -1,252 +1,306 @@
+import math
+
 import discretisedfield as df
+import micromagneticdata as mdata
 import micromagneticmodel as mm
-import numpy as np
 import pytest
 
 
 class TestZhangLi:
+    """
+    Each test defines a nanostrip with a domain wall in the left half. Current is
+    applied to move the DW to the other side of the strip. Details of the current
+    (direction, time dependence) are varied in the different tests.
+    """
+
     @pytest.fixture(autouse=True)
     def _setup_calculator(self, calculator):
         self.calculator = calculator
 
-    def setup_method(self):
-        p1 = (-5e-9, -5e-9, -3e-9)
-        p2 = (5e-9, 5e-9, 3e-9)
-        self.region = df.Region(p1=p1, p2=p2)
-        self.cell = (1e-9, 1e-9, 3e-9)
-        self.subregions = {
-            "r1": df.Region(p1=(-5e-9, -5e-9, -3e-9), p2=(5e-9, 0, 3e-9)),
-            "r2": df.Region(p1=(-5e-9, 0, -3e-9), p2=(5e-9, 5e-9, 3e-9)),
+    # inside setup_method it is not possible to use a fixture
+    @pytest.fixture(autouse=True)
+    def setup_method_as_fixture(self):
+        """Prepare nano strip (l_x=200nm) with DW at x≈70nm."""
+        p1 = (0, 0, 0)
+        p2 = (200e-9, 20e-9, 5e-9)
+        cell = (5e-9, 5e-9, 5e-9)
+        subregions = {
+            "r1": df.Region(p1=(0, 0, 0), p2=(100e-9, 20e-9, 5e-9)),
+            "r2": df.Region(p1=(100e-9, 0, 0), p2=(200e-9, 20e-9, 5e-9)),
         }
 
-    def test_scalar_scalar(self):
-        name = "zhangli_scalar_scalar"
-
-        u = 0
-        beta = 0.5
-        H = (0, 0, 1e5)
-        Ms = 1e6
+        Ms = 5.8e5
+        A = 15e-12
+        K = 0.5e6
+        anisotropy_axis = (0, 0, 1)
 
-        mesh = df.Mesh(region=self.region, cell=self.cell)
+        def init_m(p):
+            if p[0] < 70e-9:
+                return (0.1, 0.1, -1)
+            return (0.1, 0.1, 1)
 
-        system = mm.System(name=name)
-        system.energy = mm.Zeeman(H=H)
-        system.dynamics = mm.ZhangLi(u=u, beta=beta)
-        system.m = df.Field(mesh, nvdim=3, value=(0, 0.1, 1), norm=Ms)
+        system = mm.System(name="strip_x")
+        system.energy = mm.Exchange(A=A) + mm.UniaxialAnisotropy(K=K, u=anisotropy_axis)
+        mesh = df.Mesh(p1=p1, p2=p2, cell=cell, subregions=subregions)
+        system.m = df.Field(mesh, nvdim=3, value=init_m, norm=Ms)
 
-        td = self.calculator.TimeDriver()
-        td.drive(system, t=0.2e-9, n=50)
+        md = self.calculator.MinDriver()
```

Comment:
Is this okay for Mumax3?

Reply:
@samjrholt Would you prefer more local tests before merging this?